ArduCopter/land_detector.cpp

```diff
@@ -49,12 +49,9 @@ void Copter::update_land_detector()
 #if FRAME_CONFIG == HELI_FRAME
         // if rotor speed and collective pitch are high then clear landing flag
         if (motors->get_takeoff_collective() && motors->get_spool_state() == AP_Motors::SpoolState::THROTTLE_UNLIMITED) {
-#else
-        // if throttle output is high then clear landing flag
-        if (motors->get_throttle() > get_non_takeoff_throttle()) {
```

Comment:
This leaves get_non_takeoff_throttle() so that it is only used within set-mode as a check when switching to a manual throttle.  Maybe that's correct or maybe the check should be simplified to remove the helper function.

Reply:
Removing this check may increase the risk that a multicopter gets off the ground in modes with manual throttle (including Stabilize) because it means each mode becomes responsible for setting the land_complete flag.  Stabilize mode for example will only set land_complete = false if the motors library's lower_limit is false.  This means we need to be more careful that each motor library sets this flag correctly.

Reply:
Maybe we can put back this check with an additional condition that !takeoff.running().  This could close the a potential gap.

Reply:
I have done this in addition to explicitly handling leaving takeoff before it is completed.